.github/actions/detect-solidity-file-changes/action.yml

```diff
@@ -0,0 +1,36 @@
+name: 'Detect Changes Composite Action'
+description: 'Detects changes in solidity files and fails if read-only files are modified.'
+outputs:
+  changes:
+    description: 'Whether or not changes were detected'
+    value: ${{ steps.changed_files.outputs.src }}
+runs:
+  using: 'composite'
+  steps:
+
+    - name: Filter paths
+      uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+      id: changed_files
+      with:
+        list-files: 'csv'
+        filters: |
+          src:
+            - 'contracts/**/*'
+            - '.github/workflows/solidity.yml'
+            - '.github/workflows/solidity-foundry.yml'
+            - '.github/workflows/solidity-wrappers.yml'
+          read_only_sol:
+            - 'contracts/src/v0.8/interfaces/**/*'
+            - 'contracts/src/v0.8/automation/v1_2/**/*'
+            - 'contracts/src/v0.8/automation/v1_3/**/*'
+            - 'contracts/src/v0.8/automation/v2_0/**/*'
+
+    - name: Fail if read-only files have changed
+      if: ${{ steps.changed_files.outputs.read_only_sol == 'true' }}
+      shell: bash
+      run: |
+        echo "One or more read-only Solidity file(s) has changed."
+        for file in ${{ steps.changed_files.outputs.read_only_sol_files }}; do
```

Comment:
I think we might need to do something extra here since the `list-files: 'csv'` returns a comma-separated string.

I tested with this:
```
test=contracts/test/test.sol,contracts/test/test2.sol,contracts/test/test3.sol

for file in $test; do
  echo "$file was changed"
done

for file in $(echo $test | sed "s/,/ /g"); do
  echo "$file was changed"
done
```

the results came out like this:
```
contracts/test/test.sol,contracts/test/test2.sol,contracts/test/test3.sol was changed
-
contracts/test/test.sol was changed
contracts/test/test2.sol was changed
contracts/test/test3.sol was changed
```

Reply:
I haven't changed this at all. This is a copy/paste from the original detection code from https://github.com/smartcontractkit/chainlink/blob/develop/.github/workflows/solidity.yml#L23